packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -309,6 +318,13 @@ class SearchAnchor extends StatefulWidget {
   /// ```
   final BoxConstraints? viewConstraints;
 
+  /// The padding to use for the search view.
+  ///
+  /// Has no effect if the search view is full-screen.
```

Comment:
The padding is disabled if the view is full screen because it's supposed to be full screen

Reply:
>  If users don't want to use it in full-screen mode, they can set the padding to zero

This would require the users to know if the view wants to make itself fullscreen or not. The view does this based on the platform.

Reply:
> I just feel we can still use this property to make the "full-screen" view looks smaller

I agree. This is a common behavior. For example, the Viewport of a ScrollView will size itself smaller when wrapped with padding.

Reply:
AFAICT, this part of the PR is what remains to resolve before it lands.

Reply:
> I only want the padding to apply to the non-fullscreen view

Why though? I would expect the behavior to be the same in either situation. Having special cases baked in makes it harder for the user to understand what is happening and how to get the results they want. :)

Reply:
> Why though?

Because I only added the padding to have the search view not touch the bottom of the window on desktop web browsers.

> Having special cases baked in makes it harder for the user to understand what is happening and how to get the results they want. :)

There is already a special case built in to show the full screen view based on the platform... I didn't even know the view did that until it was brought up in PR review, but now that I do know (and see my app behave that way on mobile browsers) I _do not_ want the padding to apply to the full screen view.

Changing the padding if the view is full screen requires internal implementation knowledge of `SearchAnchor` which is not a good developer UX at all.